torch/csrc/autograd/python_variable.cpp

```diff
@@ -108,12 +101,21 @@ PyObject * THPVariable_Wrap(Variable var)
     Py_RETURN_NONE;
   }
 
-  if (auto obj = pyobj(var)) {
-    Py_INCREF(obj);
-    return obj;
+  c10::optional<PyObject*> mb_obj = var.unsafeGetTensorImpl()->check_pyobj(self_interpreter.get());
+  c10::impl::PyInterpreterStatus status;
+  if (mb_obj.has_value()) {
+    auto obj = *mb_obj;
+    if (obj) {
+      Py_INCREF(obj);
+      return obj;
+    }
+    // TODO: a better invariant is that if we tagged, we MUST have a valid
```

Comment:
hmm, so this is the case where the interpreter matches, but pyobj_ may have been null.  how does this happen? is it a case where multiple threads on the same interp are racing to do a wrapping?

Reply:
Nope, it's because this PR predates #56017 which guarantees that we don't deallocate the PyObject before Tensor goes dead. Prior to that PR (aka this PR), pyobj will get nulled out by THPVariable_dealloc when all Python side references to it go dead.